ha/util/conf_store.py

```diff
@@ -71,29 +73,67 @@ def get_server_pods(index):
             machine_ids.append(machine_id)
         return machine_ids
 
-    def get_cluster_cardinality(self):
+    def get_matched_labels(index: str, machine_ids: list = None) -> dict:
         """
-        Get number of nodes(pods) and their machine ids from confstore
+        Get IDs from GConf using the key node_id which gets matched with
+        cortx.io/machine-id label in
+        Args:
+        index: str = GConf index
+        machine_ids: str = list of machine ids
+        >>> Conf.get('cluster_conf', 'node>be47198d5e9191b586f9319a67dd0769>node_id')
+        '92cc6308571c4f28b7709eeca57f7352'
+        """
+        node_label_to_machine_id_map = {}
+        for machine_id in machine_ids:
+            node_id = Conf.get(index, f'node{_DELIM}{machine_id}{_DELIM}node_id')
+            # create a mapping between ID(matched with cortx.io/machine-id) and
+            # machine-id which is required for k8s monitor alert processing and alert
+            # creation.
+            node_label_to_machine_id_map[machine_id] = node_id
+        return node_label_to_machine_id_map
+
+    def get_node_name_to_machine_id_mapping(index: str, machine_ids: list = None) -> dict:
+        """
+        Get names of the Node using Gconf name field which will map to
+        cortx.io/pod-name label and then map it to actual machine-id
+        Args:
+        index: str = GConf index
+        machine_ids: str = list of machine ids
+        Ex:
+        >>> Conf.get('cluster_conf', 'node>be47198d5e9191b586f9319a67dd0769>name')
+        'cortx-data-headless-svc-ssc-vm-g4-rhev4-1599'
         """
+        name_to_macine_id_mapping = {}
+        for machine_id in machine_ids:
+            name = Conf.get(index, f'node{_DELIM}{machine_id}{_DELIM}name')
+            name_to_macine_id_mapping[name] = machine_id
+        return name_to_macine_id_mapping
 
+    def get_cluster_cardinality(self) -> (int, list, dict):
+        """
+        Get number of nodes(pods) and their machine ids from confstore
+        """
         cluster_cardinality_key = CLUSTER_CARDINALITY_KEY
         if self._confstore.key_exists(cluster_cardinality_key):
             cluster_cardinality = self._confstore.get(cluster_cardinality_key)
-            _, nodes_list = cluster_cardinality.popitem()
-            nodes_dict = json.loads(nodes_list)
+            _, nodes_info = cluster_cardinality.popitem()
+            nodes_dict = json.loads(nodes_info)
 
             #Get  number of nodes and the node list
             num_nodes = nodes_dict[CLUSTER_CARDINALITY_NUM_NODES]
-            nodes_list = nodes_dict[CLUSTER_CARDINALITY_LIST_NODES]
-            return num_nodes, nodes_list
+            label_node_dict = nodes_dict[CLUSTER_CARDINALITY_NODES]
```

Comment:
please reverse key to value and value to key

Reply:
Done.